CrmManager.php

```diff
@@ -12,25 +12,11 @@
  */
 class CrmManager
 {
-    private BazSender $client;
+    private Sender $client;
 
-    /**
-     * @var array
-     */
-    private $settings;
-
-    public function __construct(array $settings)
+    public function __construct(Sender $client)
     {
-        if (empty($settings['user'])) {
```

Comment:
Encapsulation principle violation. Context class doesn't have to know about internal logic realization